src/main/java/com/teragrep/rlp_11/RelpProbe.java

```diff
@@ -0,0 +1,254 @@
+                allSent = relpBatch.verifyTransactionAll();
+                if (!allSent) {
+                    LOGGER.warn("Transactions failed, retrying");
+                    metrics.resends.inc();
+                    relpBatch.retryAllFailed();
+                    reconnect();
+                }
+            }
+            try {
+                LOGGER.debug("Sleeping before sending next event");
+                Thread.sleep(eventDelay);
+            }
+            catch (InterruptedException e) {
+                LOGGER.warn("Sleep interrupted: <{}>", e.getMessage());
+            }
+        }
+        disconnect();
+        latch.countDown();
+        slf4jReporter.close();
+        jmxReporter.close();
+        try {
+            jettyServer.stop();
+        }
+        //CHECKSTYLE:OFF
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        //CHECKSTYLE:ON
+    }
+
+    private void connect() {
+        while (!connected && stayRunning) {
+            try (final Timer.Context context = metrics.connectLatency.time()) {
+                LOGGER.info("Connecting to <[{}:{}]>", config.getTargetHostname(), config.getTargetPort());
+                connected = relpConnection.connect(config.getTargetHostname(), config.getTargetPort());
+                LOGGER.info("Connected.");
+                metrics.connects.inc();
+            }
+            catch (TimeoutException | IOException e) {
+                LOGGER
+                        .warn(
+                                "Failed to connect to <[{}:{}]>: <{}>", config.getTargetHostname(),
+                                config.getTargetPort(), e.getMessage()
+                        );
+            }
+            if (!connected) {
+                try {
+                    LOGGER.info("Sleeping for <[{}]>ms before reconnecting", config.getReconnectInterval());
+                    Thread.sleep(config.getReconnectInterval());
+                    metrics.retriedConnects.inc();
+                }
+                catch (InterruptedException e) {
+                    LOGGER.warn("Sleep was interrupted: <{}>", e.getMessage());
+                }
+            }
+        }
+    }
+
+    private void reconnect() {
+        disconnect();
+        connect();
+    }
+
+    private void disconnect() {
+        if (!connected) {
+            LOGGER.debug("No need to disconnect, not connected");
+            return;
+        }
+        try {
+            LOGGER.info("Disconnecting..");
+            relpConnection.disconnect();
+            metrics.disconnects.inc();
+        }
+        catch (IOException | TimeoutException e) {
+            LOGGER.warn("Failed to disconnect: <{}>", e.getMessage());
+        }
+        relpConnection.tearDown();
+        LOGGER.info("Disconnected.");
+        connected = false;
+    }
+
+    public void stop() {
+        LOGGER.debug("Stop called");
+        stayRunning = false;
```

Comment:
use atomic boolean because this is interthread communication path, volatile is another option but not preferred in here.

Reply:
Fixed